Sources/FoundationExtensions/Operators+Extensions.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Operators+Extensions.swift
+//
+//  Created by Nacho Soto on 5/23/22.
+
+infix operator ???
+
+/// Equivalent to `??` but allows an `async` default value.
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+func ??? <T>(value: T?, defaultValue: @autoclosure () async throws -> T) async rethrows -> T {
```

Comment:
This won't be necessary [when Swift gets `reasync`](https://twitter.com/slava_pestov/status/1528926247828525057?s=21&t=armm7A8ik5cATfXmPQXOjg).

Reply:
maybe we can add a link here pointing to https://github.com/apple/swift-evolution/blob/main/proposals/0296-async-await.md#future-directions? That way if a dev runs into this in the future after `reasync` gets implemented, they'll know that it serves a similar purpose and can be replaced?

Reply:
Done.